eng/common/pipelines/templates/jobs/npm-release-task.yml

```diff
@@ -0,0 +1,158 @@
+              $tempDir = "$(System.DefaultWorkingDirectory)/temp_decompress"
+              New-Item -ItemType Directory -Force -Path $tempDir
+              tar -xzf $tarFile -C $tempDir
+              $packageJsonDir = "$tempDir/package/package.json"
+              $pkg = Get-Content -Raw "$packageJsonDir" | ConvertFrom-Json
+              $packageName = $pkg.Name
+              $packageVersion = $pkg.Version
+              Write-Host "npm view $packageName -json -silent"
+              $packageProps = npm view $packageName -json -silent | ConvertFrom-Json
+              $originalTags = "{}"
+              if ($packageProps.error) {
+                $LASTEXITCODE = 0
+                if ($packageProps.error.code -ne "E404") {
+                  Write-Error $packageProps.error
+                  exit 1
+                }
+              }
+              else {
+                $originalTags = $packageProps.'dist-tags' | ConvertTo-Json -Compress
+              }
+              
+              echo "##vso[task.setvariable variable=PackageName;isOutput=true]$packageName"
+              echo "##vso[task.setvariable variable=OriginalTags;isOutput=true]$originalTags"
+              echo "##vso[task.setvariable variable=IntendedTagVersion;isOutput=true]$packageVersion"
+            name: SetPackageVar
+            displayName: Get original tags
+          - task: EsrpRelease@9
+            displayName: 'Publish ${{parameters.Artifact.name}} to ESRP'
+            inputs:
+              ConnectedServiceName: 'Azure SDK PME Managed Identity'
+              ClientId: '5f81938c-2544-4f1f-9251-dd9de5b8a81b'
+              DomainTenantId: '975f013f-7f24-47e8-a7d3-abc4752bf346'
+              Usemanagedidentity: true
+              KeyVaultName: 'kv-azuresdk-codesign'
+              SignCertName: 'azure-sdk-esrp-release-certificate'
+              Intent: 'PackageDistribution'
+              ContentType: 'npm'
+              FolderLocation: ${{parameters.PathToArtifacts}}
+              Owners: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              Approvers: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
+              MainPublisher: 'ESRPRELPACMANTEST'
+              productstate: $(Tag)
+  - job: VerifyPackageTags_${{ replace(parameters.Artifact.name, '-', '_') }}
+    dependsOn:
+      - PublishPackage_${{ replace(parameters.Artifact.name, '-', '_') }}
+      - ${{ each dep in parameters.DependsOn }}:
+        - ${{ dep }}
+    condition: succeeded()
+    variables:
+      - template: /eng/pipelines/templates/variables/image.yml
+      - name: PackageName
+        value: $[dependencies.PublishPackage_${{ replace(parameters.Artifact.name, '-', '_') }}.outputs['PublishPackage_${{ replace(parameters.Artifact.name, '-', '_') }}.SetPackageVar.PackageName']]
```

Comment:
Do we need this to be per package/artifact named?

Reply:
^ as discussed in person